components/apimgt/org.wso2.carbon.apimgt.micro.gateway/org.wso2.carbon.apimgt.micro.gateway.common/src/main/java/org/wso2/carbon/apimgt/micro/gateway/common/dao/OnPremiseGatewayDAO.java

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.carbon.apimgt.micro.gateway.common.dao;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.simple.JSONArray;
+import org.wso2.carbon.apimgt.impl.utils.APIMgtDBUtil;
+import org.wso2.carbon.apimgt.micro.gateway.common.config.ConfigManager;
+import org.wso2.carbon.apimgt.micro.gateway.common.exception.OnPremiseGatewayException;
+import org.wso2.carbon.apimgt.micro.gateway.common.util.OnPremiseGatewayConstants;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+
+/**
+ * Class for Database Access
+ */
+public class OnPremiseGatewayDAO {
+    Log log = LogFactory.getLog(OnPremiseGatewayDAO.class);
+    private static final String insertIntoAPIPublishEvents = "INSERT INTO AM_API_LC_PUBLISH_EVENTS " +
+            "(TENANT_DOMAIN, API_ID, EVENT_TIME) VALUES (?, ?, ?);";
+    private static final String selectAllAPIPublishEvents = "SELECT DISTINCT API_ID FROM AM_API_LC_PUBLISH_EVENTS " +
+            "WHERE TENANT_DOMAIN = (?) AND EVENT_TIME > (?);";
+
+    public static OnPremiseGatewayDAO getInstance() {
+        return new OnPremiseGatewayDAO();
+    }
+
+    /**
+     * Adds an API publish/re-publish event to the database
+     *
+     * @param tenantDomain tenant domain
+     * @param apiId        API identifier
+     * @throws OnPremiseGatewayException OnPremiseGatewayException
+     */
+    public void addAPIPublishEvent(String tenantDomain, String apiId) throws OnPremiseGatewayException {
+        if (log.isDebugEnabled()) {
+            log.debug("Adding publish/re-publish event of API: " + apiId);
+        }
+        Connection conn = null;
+        PreparedStatement ps = null;
+        try {
+            conn = APIMgtDBUtil.getConnection();
+            conn.setAutoCommit(false);
```

Comment:
might be a goog way to get the current auto commit value and set it back in the finally block